pkg/operator/operatorstatus.go

```diff
@@ -0,0 +1,95 @@
+package operator
+
+import (
+	"fmt"
+
+	metaapi "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/util/retry"
+
+	osapi "github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+)
+
+func updateOperatorCondition(op *osapi.ClusterOperator, condition *osapi.ClusterOperatorStatusCondition) (modified bool) {
+	found := false
+	conditions := []osapi.ClusterOperatorStatusCondition{}
+
+	for _, c := range op.Status.Conditions {
+		if condition.Type != c.Type {
+			conditions = append(conditions, c)
+			continue
+		}
+		if condition.Status != c.Status {
+			modified = true
+		}
+		conditions = append(conditions, *condition)
+		found = true
+	}
+
+	if !found {
+		conditions = append(conditions, *condition)
+		modified = true
+	}
+
+	op.Status.Conditions = conditions
+	return
+}
+
+func (h *Handler) operatorStatus(condtype osapi.ClusterStatusConditionType, status osapi.ConditionStatus, msg string) error {
+	state := &osapi.ClusterOperator{
+		TypeMeta: metaapi.TypeMeta{
+			APIVersion: osapi.SchemeGroupVersion.String(),
+			Kind:       "ClusterOperator",
+		},
+		ObjectMeta: metaapi.ObjectMeta{
+			Name:      h.name,
+			Namespace: h.namespace,
+		},
+	}
+	return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		err := sdk.Get(state)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return fmt.Errorf("failed to get cluster operator resource %s/%s: %s", state.ObjectMeta.Namespace, state.ObjectMeta.Name, err)
+			}
+
+			state.Status.Conditions = []osapi.ClusterOperatorStatusCondition{
+				{
+					Type:               osapi.OperatorAvailable,
+					Status:             osapi.ConditionUnknown,
+					LastTransitionTime: metaapi.Now(),
+				},
+				{
+					Type:               osapi.OperatorProgressing,
+					Status:             osapi.ConditionUnknown,
+					LastTransitionTime: metaapi.Now(),
+				},
+				{
+					Type:               osapi.OperatorFailing,
+					Status:             osapi.ConditionUnknown,
+					LastTransitionTime: metaapi.Now(),
+				},
+			}
```

Comment:
isn't this going to reset all the conditions that aren't being explicitly set?

we should only initialize conditions to unknown if 
1) they aren't set already and
2) we're not setting them now

Reply:
oh nm, i see this is only if we didn't find the clusteroperator object.

Reply:
When we have no information and we only create a resource, the lack of status is equivalent to `unknown`. It seemed to me useful to show that we will set all the statuses. Also in this way, I hardcode a position in the array (although this is not important).

Reply:
yeah, this is fine.  It originally looked to me like you were always resetting the statuses, but I see now you're just initializing them the first time.